File: teamscale-maven-plugin/src/main/java/com/teamscale/maven/BuildVersion.java
package com.teamscale.maven;

import java.util.ResourceBundle;

/** Provides access to the Maven plugin version at runtime. */
public class BuildVersion {

	private static final ResourceBundle BUNDLE = ResourceBundle.getBundle("com.teamscale.maven.app");

	/** The version of the Teamscale Maven plugin. */
	public static final String VERSION = BUNDLE.getString("version");
}

File: impacted-test-engine/src/main/kotlin/com/teamscale/test_impacted/BuildVersion.kt
package com.teamscale.test_impacted

import java.util.*

/** Provides access to the Impacted Test Engine version at runtime. */
object BuildVersion {

    private val bundle = ResourceBundle.getBundle("com.teamscale.test_impacted.app")

    /** The version of the Teamscale Impacted Test Engine. */
    val VERSION: String = bundle.getString("version")
}
